domain/life.go

```diff
@@ -0,0 +1,14 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package domain
+
+// Life represents the life of an entity
```

Comment:
I am still struggling with this. We are still knowingly bleeding a state concept outside of the state package. You need to write utils functions to transfer from a how the DDL stores this information with primary keys into our enum types. At the moment we are saying that these int's directly correspond to how the life look up table is constructed but one change tomorrow could completely break both parts and we have no testing to cover this.

Reply:
Life is common to all state packages. Where do you propose we put it? I just want to have typed consts that state packages can use to deal with life ids

Reply:
We could incur an extra join and work with the life name strings. If that's what we'd prefer, can do that. But I didn't do that as it seems inefficient to incur that cost just to do a simple operation on a value. The argument that you shouldn't have to change code if the db relational model gets a new value for life doesn't hold either, because you'd still want to introduce a new const for the added life string value regardless. Or we could do `if foo.life == 2` using a raw int but that's not what I would call optimal.

Reply:
As discussed live, moved to a life subpackage and added a test to check for db skew.